lib/cosmos/tools/cmd_tlm_server/interface_thread.rb

```diff
@@ -89,14 +104,20 @@ def start
 
     # Disconnect from the interface and stop the thread
     def stop
+      @cancel_thread = true
+      @thread_sleeper.cancel
       @interface.disconnect
 
       if @thread
         Logger.info "Stopping packet reading for #{@interface.name}"
-        @thread.kill
+        Cosmos.kill_thread(self, @thread) if @thread != Thread.current
```

Comment:
Threads cannot gracefully kill themselves.   Calling kill_thread would timeout on the graceful part and then force kill itself.